src/cli.spec.js

```diff
@@ -300,6 +456,81 @@ describe("CLI", function () {
       });
     });
 
+    it("should return 1 if local catalog file not found", async function () {
+      return cli({
+        filename: "./testfiles/valid.json",
+        catalogs: ["./testfiles/does-not-exist.json"],
+      }).then((result) => {
+        assert.equal(1, result);
+        expect(messages.error).to.contain(
+          "ENOENT: no such file or directory, open './testfiles/does-not-exist.json'"
+        );
+      });
+    });
+
+    it("should return 1 if invalid response fetching remote catalog", async function () {
+      const mock = nock("https://example.com")
+        .get("/catalog.json")
+        .reply(404, {});
+
+      return cli({
+        filename: "./testfiles/valid.json",
+        catalogs: ["https://example.com/catalog.json"],
+      }).then((result) => {
+        assert.equal(1, result);
+        expect(messages.error[0]).to.contain(
+          "❌ Failed fetching https://example.com/catalog.json"
+        );
+        mock.done();
+      });
+    });
+
+    it("should return 1 on malformed catalog (missing 'schemas')", async function () {
+      const mock = nock("https://example.com")
+        .get("/catalog.json")
+        .reply(200, {});
+
+      return cli({
+        filename: "./testfiles/valid.json",
+        catalogs: ["https://example.com/catalog.json"],
+      }).then((result) => {
+        assert.equal(1, result);
+        mock.done();
+        expect(messages.error).to.contain(
+          "❌ Malformed catalog at https://example.com/catalog.json",
+        );
+      });
+    });
+
+    it("should return 1 on malformed catalog ('schemas' should be an array)", async function () {
+      const mock = nock("https://example.com")
+        .get("/catalog.json")
+        .reply(200, { schemas: {} });
+
+      return cli({
+        filename: "./testfiles/valid.json",
+        catalogs: ["https://example.com/catalog.json"],
+      }).then((result) => {
+        assert.equal(1, result);
+        mock.done();
+        expect(messages.error).to.contain(
+          "❌ Malformed catalog at https://example.com/catalog.json"
+        );
+      });
+    });
+
+    it("should return 1 on malformed catalog ('schemas' elements should contains a valid url)", async function () {
+      return cli({
+        filename: "./testfiles/valid.json",
+        catalogs: ["./testfiles/catalog-local.json"],
+      }).then((result) => {
+        assert.equal(1, result);
+        expect(messages.error).to.contain(
+          "❌ Malformed catalog at ./testfiles/catalog-local.json"
+        );
+      });
+    });
```

Comment:
Nice :+1:  This block of tests for the various failure/invalid cases are very comprehensive :)